src/sentry/tasks/weekly_reports.py

```diff
@@ -647,7 +647,17 @@ def fetch_key_performance_issue_groups(ctx):
 def deliver_reports(ctx, dry_run=False, target_user=None, email_override=None):
     # Specify a sentry user to send this email.
     if email_override:
-        send_email(ctx, target_user, dry_run=dry_run, email_override=email_override)
+        try:
+            send_email(ctx, target_user.id, dry_run=dry_run, email_override=email_override)
+        except AttributeError:
+            logger.exception(
+                "deliver_reports.email_override.error",
+                extra={
+                    "dry_run": dry_run,
+                    "target_user": json.dumps(target_user),
```

Comment:
Either it is JSON serializable, in which case logging handler will take care of serialization, so no need to `json.dumps` here, or it's not JSON serializable, and it's going to throw serialization exception here. 

FWIW, I'm adding a fix to specifically suppress JSON serialization errors in logging, this would bypass that fix. See: https://github.com/getsentry/sentry/pull/60131

Reply:
oh awesome fix!! thanks, i'll remove the dumps then 🥔